actors.md

```diff
@@ -580,6 +580,38 @@ func SubmitPost(proofs []PoStProof, faults []FaultSet, recovered BitField, done
 	miner.NextDoneSet = done
 	miner.ArbitratedDeals.Clear()
 }
+
+func ValidateFaultSets(faults []FaultSet, recovered, done BitField) bool {
```

Comment:
Checking for understanding: do methods like `IsSubsetOf`, `Intersects` and `Combine` actually operate on the rle+ Bitfield encoding?  I've heard that this encoding is not designed for efficient updates, it is good only at being compact.  

I am guessing that there is an implicit step in `submitPoSt` where the state machine translates the efficiently encoded BitField into an internal representation that can quickly do unions and intersections.  If this is correct is this hypothesized state-machine-memory-resident type in scope of being specced out?  Also if this is correct is it the case that we should be concerned about *its* performance properties?

Hopefully I am just overthinking this...

Reply:
Yeah, i'm not necessarily getting into implementation details, this is more of an abstract operation. You have a set of items, these are well known operations on sets of items.

That said, you should be able to operate fairly efficiently over the encoded bitsets, combining multiple rle+ bitfields can be done in a single O(n) pass (where n is the number of bit runs). If the implementation decides to parse that into a different structure internally for performance reasons, I think thats fine.

Reply:
@dignifiedquire and I have discussed plumbing a "current block height" value to the PoSt-generation routine (which is updated when the node learns of a new heaviest tipset). @whyrusleeping - Is it your expectation that a single call to `generate_post` could produce multiple `FaultSet`s (each with a different block height)? 

Also: Is it your expectation that a non-empty `FaultSet` produced during the miner's first proving period would be included in a `submitPoSt` message created during a subsequent proving period?

Reply:
@ZenGround0 

> I am guessing that there is an implicit step in submitPoSt where the state machine translates the efficiently encoded BitField into an internal representation that can quickly do unions and intersections.

This too is what I assumed. I would have thought we'd use RLE+ when converting `BitField` to `[]byte` and back - and that inside the state machine we'd be working with some fancy type optimized for set operations.

Reply:
> Is it your expectation that a single call to generate_post could produce multiple FaultSets (each with a different block height)?

Yes, the failure could happen part ways through the post. multiple failures could happen at different times during the PoSt

> Also: Is it your expectation that a non-empty FaultSet produced during the miner's first proving period would be included in a submitPoSt message created during a subsequent proving period?

No, at the end of a proving period, any sectors that are not marked as recovered are dropped from the sector set and should not be proved during the next PoSt.